### PR TITLE
HSEARCH-3554 The Elasticsearch backend defaults to a 10 element hit limit

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net//browse/HSEARCH-3554

### CODE REVIEW

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/impl/ElasticsearchIndexSearchQuery.java [hunk 1]
```diff
@@ -27,6 +27,11 @@
  */
 public class ElasticsearchIndexSearchQuery<T> implements IndexSearchQuery<T> {
 
+	/**
+	 * ES default limit for (limit + offset); any search query beyond that limit will be rejected.
+	 */
+	private static final long MAX_RESULT_WINDOW_SIZE = 10000;
```
[the remark this answers is not in the document]
Comment (yrodiere) [reply backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/impl/ElasticsearchIndexSearchQuery.java]: I knew you'd say that... https://hibernate.atlassian.net/browse/HSEARCH-3556 :)

[the remark this answers is not in the document]
Comment (gsmet) [reply backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/impl/ElasticsearchIndexSearchQuery.java]: OK, cool, I agree it should be done in a second step.

**Comment by yrodiere** [comment]
> we should throw an error if we have more results than the limit. But I'm not sure we can have the information?

We might have this information, but I have three problems with the idea:

* I think it could hurt more than it helps, especially in a production environment. A warning might be more suitable.
* I'm not sure we'll be able to know this reliably in ES7 with the optimizations they added regarding hit count.
* If someone retrieves 10,000 results all in one block, they probably have another issue... ? Is it really worth spending our time on this?

I'll add something to HSEARCH-3556: we can issue a warning when this happens and the max window size is not explicitly set by the users (= he may not know about this limitation).

**Comment by gsmet** [comment]
Yeah, I wrote "error" but I was thinking "warning". Agreed on the rest.

**Pull request merged**